production/helm/loki/templates/monitoring/metrics-instance.yaml

```diff
@@ -0,0 +1,30 @@
+{{- if .Values.monitoring.serviceMonitor.enabled }}
+{{- with .Values.monitoring.serviceMonitor.metricsInstance }}
+apiVersion: monitoring.grafana.com/v1alpha1
+kind: MetricsInstance
+metadata:
+  name: {{ include "loki.fullname" $ }}
+  namespace: {{ .namespace | default $.Release.Namespace }}
+  {{- with .annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  labels:
+    {{- include "loki.labels" $ | nindent 4 }}
+    {{- with .labels }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  {{- with .remoteWrite}}
+  remoteWrite:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+
+  # Supply an empty namespace selector to look in all namespaces. Remove
+  # this to only look in the same namespace as the MetricsInstance CR
+  serviceMonitorNamespaceSelector: {}
```

Comment:
do we need this? what happens if omitted, I would assume the default is same namespace? alternatively we know the namespace, so we could also be explicit?

Reply:
Ok, let's use `.Release.Namespace`

Reply:
loki namespace is in `.Release.Namespace`

Reply:
@trevorwhitney turns out we can only match on labels of the namespace (https://github.com/prometheus-operator/prometheus-operator/issues/1331#issuecomment-388917575). I'd say we move on without a selector and follow up.

Reply:
since we're not creating the namespace resources ourselves we can't control what labels they're created with, so I agree with leaving this empty for now.